custom_components/ha_hatch/switch.py

```diff
@@ -43,3 +45,23 @@ def turn_on(self, **kwargs):
 
     def turn_off(self, **kwargs):
         self.rest_device.set_on(False)
+
+class HatchToddlerLockSwitch(RestEntity, SwitchEntity):
+    _attr_device_class = SwitchDeviceClass.SWITCH
+    _attr_icon = "mdi:human-child"
+
+    def __init__(self, rest_device: RestPlus):
+        super().__init__(rest_device, "Toddler Lock")
+
+    def _update_local_state(self):
+        if self.platform is None:
+            return
+        _LOGGER.debug(f"updating state:{self.rest_device}")
+        self._attr_is_on = self.rest_device.toddler_lock
+        self.async_write_ha_state()
+
+    def turn_on(self, **kwargs):
+        self.rest_device.set_toddler_lock(True, "never")
```

Comment:
Will need to update this after https://github.com/dahlb/hatch_rest_api/pull/11 merges.

Reply:
This has been updated.